scripts/codex_autoreview.py

```diff
@@ -0,0 +1,641 @@
+    req.add_header("Authorization", f"Bearer {token}")
+    req.add_header("Accept", accept)
+    req.add_header("X-GitHub-Api-Version", "2022-11-28")
+    if body is not None:
+        req.add_header("Content-Type", "application/json")
+    # Debug request
+    if _debug_level() >= 2:
+        preview = None
+        if data is not None:
+            try:
+                preview = json.dumps(data)[:800]
+            except Exception:
+                preview = str(data)[:800]
+        _dbg(2, f"HTTP {method} {url} accept={accept} data={preview}")
+
+    # Dry-run support for POSTs
+    if os.environ.get("DRY_RUN") == "1" and method.upper() == "POST":
+        _dbg(1, f"DRY_RUN: skipping POST {url}")
+        return {"dry_run": True, "url": url, "data": data}
+
+    try:
+        with urlopen(req) as resp:  # nosec - GitHub API
+            ctype = resp.headers.get("Content-Type", "")
+            code = getattr(resp, "status", None) or resp.getcode()
+            raw = resp.read()
+            if _debug_level() >= 2:
+                _dbg(2, f"HTTP {method} {url} -> {code} ct={ctype} len={len(raw)}")
+            if "application/json" in ctype:
+                return json.loads(raw.decode("utf-8"))
+            return raw
+    except HTTPError as e:  # surface error payload for debugging
+        try:
+            raw = e.read()
+            msg = raw.decode("utf-8")
+            print(f"GitHub API error body: {msg}", file=sys.stderr)
+        except Exception:
+            pass
+        raise
+
+
+def _parse_valid_head_lines_from_patch(patch: str) -> set[int]:
+    """Return the set of HEAD-side line numbers present in a unified diff patch.
+
+    We treat context (' ' prefix) and additions ('+') as valid inline-anchorable
+    lines on the RIGHT side. Deletions ('-') are not valid on the head side.
+    """
+    valid: set[int] = set()
+    i_old = i_new = 0
+    for line in patch.splitlines():
+        if line.startswith("@@"):
+            # Example: @@ -12,7 +12,9 @@ optional section
+            try:
+                header = line.split("@@")[1].strip()
+            except Exception:
+                header = line
+            try:
+                minus, plus = header.split(" ")[0:2]
+            except Exception:
+                # Fallback: find + segment
+                parts = header.split()
+                minus = parts[0]
+                plus = parts[1] if len(parts) > 1 else "+0,0"
+            # Parse -a,b and +c,d
+            try:
+                plus_nums = plus[1:].split(",")
+                i_new = int(plus_nums[0]) - 1  # will pre-increment below
+                minus_nums = minus[1:].split(",")
+                i_old = int(minus_nums[0]) - 1
+            except Exception:
+                i_new = i_old = 0
+            continue
+
+        if not line:
+            continue
+        tag = line[0]
+        if tag == " ":
+            i_old += 1
+            i_new += 1
```

Comment:
[P3] PR body value is read but discarded when composing the prompt

In _compose_prompt (scripts/codex_autoreview.py) the code calls `pr.get("body") or ""` without assigning it to a variable, so the PR body is not included in the composed prompt. The lines `pr_title = pr.get("title") or ""` followed by `pr.get("body") or ""` indicate an apparent oversight; the PR body should be captured and included in the prompt if intended. This is a correctness/verbosity issue affecting prompt completeness, and should be fixed by assigning the body to a variable and including it in the prompt context.